packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -545,6 +615,33 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
     this.securityGroups.push(securityGroup);
   }
 
+  /**
+   * Emit all group metrics, these metrics describe the group rather than any of its instances
+   * see all available types https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-instance-monitoring.html#as-group-metrics
+   * if granularity is not provided, 1Minute will be set
+   * @default - disabled
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-metricscollection.html
+   */
+  public emitAllMetricsCollections(options?: MetricsCollectionOptions) {
```

Comment:
**Future Proofing Alert** 
Given that the only available property is `granularity`, this is indeed future proofing.
The justification for this is that  group metrics are CloudWatch metrics, and  it's probably safe to assume that in the future other metrics properties, such as period and statistics will be added. If the methods accepts individual properties, this might its signature long in the future.

Reply:
I disagree. The better way is to not have no parameters to this method. 
As and when new properties are needed, they can be added in here as optional properties, including `MetricsCollectionOptions`. The future proofing strategy you have take here is probably premature and can be avoided.

Reply:
Are referring to the fact the `granularity` has only one possible value?

Reply:
Yes, correct. There's no value in modeling Granularity in the CDK when it has only one possible value. We just hardcode it in the CDK and reduce the number of things the user has to configure.
When there is a second possible value in the future, it can be modeled into the CDK, and it will not be a breaking change.